java-manta-client-unshaded/src/main/java/com/joyent/manta/http/MantaHttpRequestRetryHandler.java

```diff
@@ -68,10 +68,14 @@ public boolean retryRequest(final IOException exception,
             return false;
         }
 
-        if (logger.isDebugEnabled() && executionCount <= getRetryCount()) {
-            String msg = String.format("Request failed, %d/%d retry.",
-                    executionCount, getRetryCount());
-            logger.debug(msg, exception);
+        if (logger.isDebugEnabled()) {
+            if (NON_RETRIABLE.contains(exception)) {
```

Comment:
From this error: 

`
Argument 'exception' should not be passed to this method; its type IOException is not compatible with its collection's type argument Class<? extends IOException>
`

This line should be:

```
NON_RETRIABLE.contains(exception.getClass())
```

or accounting for inheritance, something like

```
NON_RETRIABLE.stream().anyMatch(exClass -> exception instanceof exClass)
```